### PR TITLE
fix(prefetch): Fix the inclusion of `@types/network-information`

### PR DESCRIPTION
**Pull request by connor-baer**
## Changes

https://github.com/withastro/astro/pull/7104 added the `"files"` field to all packages to limit the files that are published to npm. The `@astro/prefetch` package only includes the `dist` folder, however, the code tries to import from the `@types` folder, causing the following error:

```
Could not resolve "../@types/network-information.d.ts" from "node_modules/@astrojs/prefetch/dist/client.js"
```

~I added the `@types` folder to the `"files"` field to fix it.~

I changed the import to a triple-slash directive as [suggested](https://github.com/withastro/astro/pull/7123#issuecomment-1554172407) by @bluwy.

## Testing

I did not test the change as it seems straightforward.

## Docs

This fixes an expected behavior, so an update to the docs doesn't seem necessary.

### CODE REVIEW

**Comment by bluwy** [comment]
Hi, thanks for catching this. I didn't expect the types file to be imported, and it doesn't seem right that it's imported by a JS file. 

I think instead we can remove the import here:
https://github.com/withastro/astro/blob/31cbf4357e866abab62bb4ccf949f5a59aec06ac/packages/integrations/prefetch/src/client.ts#L2
and add this to the top of the file instead.
```js
/// <reference types="../@types/network-information.d.ts" />
```

**Comment by connor-baer** [comment]
@bluwy That makes sense! I've changed the approach as suggested.

File: .changeset/brown-goats-serve.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+
+---
+'@astrojs/prefetch': patch
+---
+
+Fix the inclusion of `@types/network-information`.
```
Comment (bluwy) [inline .changeset/brown-goats-serve.md:6]:
```suggestion
---
'@astrojs/prefetch': patch
---

Fix the inclusion of `@types/network-information`.

```

small formatting tweak so i hope it doesn't trip up changesets

**Comment by millette** [comment]
Are released automated/on a schedule? Downgrading for now.

**Comment by bluwy** [comment]
I've cut a new release to fix it.

**Pull request merged**